Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,135 @@
-public struct SimpleCurrencyConverter {
-    public private(set) var text = "Hello, World!"
+import Foundation
 
-    public init() {
+public enum Currency: String {
+    case JPY = "JPY"
+    case USD = "USD"
+    case GBP = "GBP"
+    case EUR = "EUR"
+}
+
+public class SimpleCurrencyConverter: CurrencyConverterProtocol {
+
+    private var exchangeRatesAPIKey: String = ""
+    private var networkManager: ExchangeRatesNetworkManager?
+    private static var sharedInstance: CurrencyConverterProtocol?
+    
+    /// Public setup singleton instance method for a class that allows for the setting of an exchange rates API key and a URL session.
+    ///
+    /// If an exchange rates API key is provided, it is set to the class's exchangeRatesAPIKey property.
+    /// If no exchange rates API key is provided, the class attempts to retrieve an API key from a plist file using the ExchangeRatesAPIKey keyword.
+    ///
+    /// If a URL session is provided, it is used to initialize the class's network manager.
+    /// If no URL session is provided, the class's network manager is initialized with the shared URL session.
+
+    public static func shared(exchangeRatesAPIKey: String? = nil, urlSession: URLSession? = nil) -> CurrencyConverterProtocol {
```

Comment:
There's one problem with this method. I can call `shared(exchangeRatesAPIKey: "abc")` for the first time and then `shared(exchangeRatesAPIKey: "123")`. The second (and any subsequent) configuration will be ignored. 
If you want to keep singleton instance then I think you need to have a separate function for setup/configure and a var for accessing shared instance. OR you can just return to non-singleton approach

Reply:
I made a little improvement in the initialization. Added separated access to shared instance and setup methods. The setup method initializes the shared instance object which will be returned using the shared instance. The second (and any subsequent) setup method call will rewrite a shared instance with a new one containing new key or session values.